parallel.go

```diff
@@ -298,3 +304,137 @@ func (w *parallelManager) Call(data interface{}) error {
 func (w *parallelManager) Wait() (interface{}, error) {
 	return nil, ErrWouldDeadlock
 }
+
+// goWorker is an implementation of the Worker interface that operates
+// in a parallel fashion, utilizing a semaphore to control the number
+// of goroutines allowed to operate at once.
+type goWorker struct {
+	sync.Mutex
+	state  pState              // State of the worker
+	serial *sync.Mutex         // A mutex for serializing Runner.Integrate
+	runner Runner              // The runner to be invoked by the workers
+	gonner *sync.Once          // A once incarnation for getting the result
+	result interface{}         // The result from the work
+	limit  *semaphore.Weighted // Semaphore to limit concurrent execution
+	wg     *sync.WaitGroup     // Wait group to use for waits
+}
+
+// NewGoWorker constructs a worker utilizing a semaphore to limit
+// concurrency of worker goroutines.  Go workers can receive Call and
+// Wait invocations from any goroutine.  A go worker is initialived
+// with a desired maximum number of simultaneously executing
+// goroutines; if that number is less than or equal to 0, no limit is
+// enforced on the number of simultaneous goroutines.
+func NewGoWorker(runner Runner, workers int) Worker {
+	// Initialize a semaphore
+	var sem *semaphore.Weighted
+	if workers > 0 {
+		sem = semaphore.NewWeighted(int64(workers))
+	}
+
+	return &goWorker{
+		serial: &sync.Mutex{},
+		runner: runner,
+		gonner: &sync.Once{},
+		limit:  sem,
+		wg:     &sync.WaitGroup{},
+	}
+}
+
+// work is a helper that executes in a fresh goroutine.  It acquires
+// the semaphore, executes the runner's Run method with the desired
+// data, runs the runner's Integrate method with the result, then
+// dones the wait group.
+func (w *goWorker) work(data interface{}) {
+	// Signal done when we're done
+	defer w.wg.Done()
+
+	// First, acquire the semaphore; this limits the parallelism
+	if w.limit != nil {
+		w.limit.Acquire(context.Background(), 1) // cannot error with background context
+	}
+
+	// Now we can run the runner
+	result := panicer(w.runner.Run, data)
```

Comment:
One comment about variable spelling is pedantic and a further comment about panicer versus panciker is even pedanticker